motionplan/motionPlanner.go

```diff
@@ -511,6 +516,41 @@ func CheckPlan(
 		return err
 	}
 
+	// create a list of segments to iterate through
+	var segments []*ik.Segment
+	if relative {
+		segments = make([]*ik.Segment, 0, len(poses)+1)
+
+		// get the inputs we were partway through executing
+		checkFrameGoalInputs, ok := plan.Trajectory()[0][checkFrame.Name()]
+		if !ok {
+			return errors.New("could not get inputs for checkFrame")
+		}
+
+		// get checkFrame's currentInputs
+		checkFrameCurrentInputs, ok := currentInputs[checkFrame.Name()]
+		if !ok {
+			return errors.New("could not get inputs for checkFrame")
+		}
+
+		// calculate pose with respect to checkFrame's coordinate system
+		relativeCurrentPosition, err := checkFrame.Transform(checkFrameCurrentInputs)
+		if err != nil {
+			return err
+		}
+
+		// pre-pend to segments so we can connect to the input we have not finished actuating yet
+		segments = append(segments, &ik.Segment{
```

Comment:
This has the same bug that was present before Ray's recent PR.

Look below at like 569. Our start configuration is overwritten; this will check the currently executing trajectory starting at 0, *not* starting at `checkFrameCurrentInputs` as desired.